tests/companies/factories.py

```diff
@@ -155,9 +156,18 @@ class Params:
     post_code = factory.LazyFunction(create_fake_postcode)
     city = factory.Faker("city", locale="fr_FR")
     source = models.Company.SOURCE_ASP
-    convention = factory.SubFactory(SiaeConventionFactory, kind=factory.SelfAttribute("..kind"))
     department = factory.LazyAttribute(lambda o: department_from_postcode(o.post_code))
 
+    @factory.post_generation
+    def with_convention(self, create, extracted="default", **kwargs):
+        if create is False or extracted is False:
+            return
+
+        if self.is_subject_to_iae_rules:
```

Comment:
```suggestion
        if self.is_subject_to_iae_rules and not self.convention:
```
Sinon cela donne:
```
In [19]: c = SiaeConventionFactory()

In [20]: company = CompanyFactory(kind='EI', convention=c)

In [21]: company.convention == c
Out[21]: False
```
🙈

Reply:
Je me demande du coup si un:

```
    @factory.post_generation
    def convention(self, create, extracted="default", **kwargs):
```
marcherait et ne serait pas plus clair ?